### PR TITLE
fix train val create in current project

### PR DESCRIPTION
**Pull request by typhoonzero**
Try create train and validation temporary tables in current database or MaxCompute project, because we may not have write access to the database where the training table located in.

### CODE REVIEW

File: sql/create_train_val.go [hunk 1]
```diff
@@ -156,13 +156,20 @@ func createTable(database, table, origin string, db *DB, cond string) (string, e
...
 func namingTrainAndValDataset(origTable string) *trainAndValDataset {
-	// hive returns a table with a database name
-	flattenTbl := strings.Replace(origTable, ".", "__", -1)
+	// NOTE: We always create temporary table under current in using
+	// database or MaxCompute project, because the origin table may
+	// from another database which we may not have write access.
+	var origTableWithNoDBPrefix string
+	origTableParts := strings.Split(origTable, ".")
+	// FIXME(typhoonzero): use last part, if len != 2 should return some error
+	origTableWithNoDBPrefix = origTableParts[len(origTableParts)-1]
```
Comment (weiguoz) [inline sql/create_train_val.go]: Remove `var origTableWithNoDBPrefix string` ?
Tiny suggestion: how about renaming `origTableWithNoDBPrefix` to `origTableWithoutDBPrefix `

File: sql/create_train_val.go [hunk 2]
```diff
@@ -156,13 +156,20 @@ func createTable(database, table, origin string, db *DB, cond string) (string, e
...
 
 func namingTrainAndValDataset(origTable string) *trainAndValDataset {
-	// hive returns a table with a database name
-	flattenTbl := strings.Replace(origTable, ".", "__", -1)
+	// NOTE: We always create temporary table under current in using
+	// database or MaxCompute project, because the origin table may
+	// from another database which we may not have write access.
+	var origTableWithNoDBPrefix string
+	origTableParts := strings.Split(origTable, ".")
+	// FIXME(typhoonzero): use last part, if len != 2 should return some error
```
Comment (weiguoz) [inline sql/create_train_val.go]: I'm not sure if we can make this assertion. Is it possible the `len` equals to 1?

Comment (typhoonzero) [reply sql/create_train_val.go]: We must support user write sqls like `select * from myproject.mytabe train ...`

**Review by weiguoz — DISMISSED** [review]
LGTM.

File: sql/create_train_val.go (hunk 2 above)
Comment (typhoonzero) [reply sql/create_train_val.go]: Will update the comment, thanks.

**Review by weiguoz — APPROVED** [review]

**Pull request merged**